exploration/0470-open-close-expressions.md

```diff
@@ -0,0 +1,222 @@
+- As a translator or tool, I want to be able to segment larger messages or process messages "segmented" from larger blocs of text, such that markup is split between messages or segments.
+
+- As a translator, I want my tools to be able to leverage translations where the text differs only in markup, e.g.:
+
+  > ```
+  > {This is {+b}sure{-b} good.}
+  > {This is {+i}sure{-i}good.}
+  > ```
+
+- As a CAT tool, I want to use the concepts of open, close, and standalone that I am already familiar with, to provide certain functionalities above.
+
+- As a developer, I want to connect message markup with existing element instances in the UI.
+
+- As a developer, I want to sanitize translations without removing the markup important to the message.
+
+  > Source HTML:
+  >
+  > ```
+  > <popup-info img="img/alt.png" data-text="Your card validation code (CVC) is an extra security feature — it is the last 3 or 4 numbers on the back of your card."></popup-info>
+  > ```
+  >
+  > Message:
+  >
+  > ```
+  > {+popup-info img=|img/alt.png| data-text=|Your card validation code (CVC) is an extra security feature — it is the last 3 or 4 numbers on the back of your card.|}{-popup-info}
+  > ```
+
+**_Non-markup use cases to consider (which may or may not be addressed by the design)_**
+
+- Represent templating language constructs so that they can pass through
+  the translation process and are not exposed to translation. For example,
+  Mustache templates.
+
+- Represent translation metadata, such as XLIFF placeholders, so that
+  portions of a message are "protected" or otherwise hinted in the
+  translation process.
+
+## Requirements
+
+Be able to indicate that some identified markup applies to
+a non-empty sequence of pattern parts (text, expressions).
+
+Markup spans may be nested,
+as in `<b>Bold and <i>also italic</i></b>`.
+
+Markup may have localisable options,
+such as `Click <a title="Link tooltip">here</a> to continue`.
+
+As with function options,
+markup options should be defined in a registry so that they can be validated.
+
+## Constraints
+
+Due to segmentation,
+markup may be split across messages so that it opens in one and closes in another.
+
+Following previously established consensus,
+the resolution of the value of an _expression_ may only depend on its own contents,
+without access to the other parts of the selected pattern.
+
+## Proposed Design
+
+This design relies on the recognition that the formatted output of MF2
+may be further processed by other tools before presentation to a user.
+
+Let us add _markup_ as a new type of _placeholder_,
+in parallel with _expression_:
+
+```abnf
+pattern = "{" *(text / placeholder) "}"
+placeholder = expression / markup
+
+markup       = "{" [s] markup-body [s] "}"
+markup-body  = (markup-standalone *(s option))
+             / (markup-open *(s option))
+             / markup-close
+markup-standalone = "#" name
+markup-open       = "+" name
+markup-close      = "-" name
```

Comment:
These sigils will need further consideration before we eventually accept this proposed design. As we evolve the syntax, `#` may get used for statements, and some other alternative open/close pair may end up being preferred.